### PR TITLE
system variables: add system variable cte_max_recursion_depth

### PR DESCRIPTION
**Pull request by xiongjiwei**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
@xiongjiwei Please involve technical review(s). If this content applies to v5.1 (sprint3), please add the v5.1 label. Thanks!

File: system-variables.md [hunk 1]
```diff
@@ -1236,3 +1236,9 @@ set tidb_slow_log_threshold = 200;
 - 作用域：SESSION | GLOBAL
 - 默认值：ON
 - 这个变量用于控制计算窗口函数时是否采用高精度模式。
+
+### `cte_max_recursion_depth`
+
+- 作用域：SESSION | GLOBAL
+- 默认值：1000
+- 这个变量用于控制公共表表达式最大递归深度。
```
Comment (TomShawn) [inline system-variables.md:1244]:
```suggestion
- 这个变量用于控制公共表表达式的最大递归深度。
```

**Comment by TomShawn** [comment]
@xiongjiwei 能按照字母顺序放在前面吗？

**Review by wjhuang2016 — COMMENTED** [review]
LGTM

**Comment by xiongjiwei** [comment]
> @xiongjiwei 能按照字母顺序放在前面吗？

updated. PTAL

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**